### PR TITLE
build: support ccache and sccache in the Makefile build (#170)

### PR DESCRIPTION
**Pull request by isaac-io**
Add support for detecting and using ccache/sccache to the Makefile build in order to speed up build times as it's done in the CMake configuration.

### CODE REVIEW

File: build_tools/build_detect_platform [hunk 1]
```diff
@@ -102,6 +102,12 @@ if test -z "$AR"; then
     fi
 fi
 
+if command -v sccache > /dev/null; then
+    CCACHE=sccache
+elif command -v ccache > /dev/null; then
+    CCACHE=ccache
+fi
+
```
Comment (mrambacher) [inline build_tools/build_detect_platform:110]: Please provide a way to turn this on/off.  I have spent too long on issues fighting with the build system to clean up the cache builds.

**Review by mrambacher — COMMENTED** [review]
There needs to be some switch to enable/disable the use of the cache.  I would prefer if (at least initially) the default is OFF until it can be generally tested and validated by the community.

Does similar work also need to be implemented for CMake or is that already in place?

**Comment by isaac-io** [comment]
> There needs to be some switch to enable/disable the use of the cache. I would prefer if (at least initially) the default is OFF until it can be generally tested and validated by the community.
> 
> Does similar work also need to be implemented for CMake or is that already in place?

I did it this way (without a way to turn it off) because that's how it's implemented on the CMake side. I don't mind adding a switch here, but I'd prefer not to do that for the CMake build because it would change the existing behaviour.

**Review by mrambacher — APPROVED** [review]
LGTM!

**Pull request merged**